projects/packages/connection/src/class-tokens.php

```diff
@@ -0,0 +1,592 @@
+<?php
+/**
+ * The Jetpack Connection Tokens class file.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+use Automattic\Jetpack\Constants;
+use Automattic\Jetpack\Roles;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * The Jetpack Connection Tokens class that manages tokens.
+ */
+class Tokens {
+
+	const MAGIC_NORMAL_TOKEN_KEY = ';normal;';
+
+	/**
+	 * Deletes all connection tokens and transients from the local Jetpack site.
+	 *
+	 * @return bool True if disconnected successfully, false otherwise.
+	 */
+	public function delete_all() {
```

Comment:
Ideally I'd only leave:

```
Jetpack_Options::delete_option(
			array(
				'blog_token',
				'user_token',
				'user_tokens',
			)
		);
```
then:
`Jetpack_Options::delete_raw_option( 'jetpack_secrets' )` could go to a corresponding method in Secrets.

And all the rest logic stay in `Manager`. By rest logic, I'd define:
- Checking for existing filters
- Removing the `master_user`, `time_diff` and `fallback_no_verify_ssl_secrets`
- Removing cached connected user data
- Deleting XML-RPC errors

Maybe the plan is to take care of all these in a follow-up PR?

Reply:
Fixed, moved the stuff around as you suggested

Reply:
Thanks for being so open to these suggestions. Appreciate it 👍